### PR TITLE
Allow access to AttributeFamily in DataAccessor

### PR DESCRIPTION
**Pull request by LesTR**
`DataAccessor` should have access to `AttributeFamilyDescriptor` not just `EntityDescriptor` and configuration.

### CODE REVIEW

File: core/src/main/java/cz/o2/proxima/storage/internal/AbstractDataAccessorFactory.java [hunk 1]
```diff
@@ -69,6 +70,29 @@ default void setup(Repository repo) {}
    * @param uri the URI to create accessor for
    * @param cfg optional additional configuration
    * @return {@link AbstractDataAccessor} for given operator and family
+   * @deprecated replaced with {@link AbstractDataAccessorFactory#createAccessor(DataOperator,
+   *     AttributeFamilyDescriptor)}
    */
-  T createAccessor(OP operator, EntityDescriptor entity, URI uri, Map<String, Object> cfg);
+  @Deprecated
+  default T createAccessor(OP operator, EntityDescriptor entity, URI uri, Map<String, Object> cfg) {
```
Comment (je-ik) [inline core/src/main/java/cz/o2/proxima/storage/internal/AbstractDataAccessorFactory.java:77]: I think we are free to do this breaking change. This tends to do more harm than good, over time.

**Review by je-ik — COMMENTED** [review]
+1 overall.
I think we should simply remove the old `createAccessor` without deprecating it. API stability will be a concern once we move closer to `1.0.0`.

File: core/src/main/java/cz/o2/proxima/storage/internal/AbstractDataAccessorFactory.java (hunk 1 above)
Comment (LesTR) [reply core/src/main/java/cz/o2/proxima/storage/internal/AbstractDataAccessorFactory.java:77]: That will require a lot of changes in existed accessors which I'll do later in https://github.com/O2-Czech-Republic/proxima-platform/issues/192

**Review by je-ik — APPROVED** [review]

File: core/src/test/java/cz/o2/proxima/storage/internal/AbstractDataAccessorFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import cz.o2.proxima.storage.internal.AbstractDataAccessorFactory.Accept;
+import cz.o2.proxima.util.Optionals;
+import cz.o2.proxima.util.TestUtils;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+import org.junit.Test;
+
+// To make sonar happy...
```
Comment (dmvk) [inline core/src/test/java/cz/o2/proxima/storage/internal/AbstractDataAccessorFactoryTest.java:33]: @LesTR ...

**Pull request merged**